### PR TITLE
refactor(ulid): improve error types and messages

### PR DESCRIPTION
**Pull request by iuioiua**

### CODE REVIEW

File: ulid/mod.ts [hunk 1]
```diff
@@ -84,12 +85,14 @@ export function decodeTime(ulid: string): number {
     .reduce((carry, char, index) => {
       const encodingIndex = ENCODING.indexOf(char);
       if (encodingIndex === -1) {
-        throw new Error("invalid character found: " + char);
+        throw new SyntaxError(`Invalid ULID character found: ${char}`);
```
Comment (kt3k) [inline ulid/mod.ts:88]: I've never seen a library throwing a SyntaxError for non-javascript syntax errors. I think this should be TypeError or Error.

Comment (iuioiua) [reply ulid/mod.ts:88]: Reasoning based on https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Errors/JSON_bad_parse

Comment (kt3k) [reply ulid/mod.ts:88]: Why the above one being an `Error` and this one being `SyntaxError`?

Comment (iuioiua) [reply ulid/mod.ts]: Hm... Yeah, they should both be the same error type... Changed to `Error`.

**Review by kt3k — APPROVED** [review]
LGTM

**Pull request merged**